src/lib/SideBar/SideBarMenuItem.tsx

```diff
@@ -11,10 +11,10 @@ interface SideBarMenuItemProps {
 }
 
 const SideBarMenuItem = ({ item, depth = 0 }: SideBarMenuItemProps) => {
-  const { toggleItem, LinkRenderer } = useSideBarMenuContext();
+  const { toggleItem, LinkRenderer, expandedMenuItemIds } = useSideBarMenuContext();
 
   const hasChildren = (item.children?.length || 0) > 0;
-  const isOpen = item?.expanded === true;
+  const isOpen = item.expanded === true || expandedMenuItemIds.includes(item.id);
```

Comment:
with this logic, if the state thinks an item is expanded, then you cannot close it from the outside...
maybe `item.expanded ?? expandedMenuItemIds.includes(item.id);` could be a solution? so it takes the expanded if set, else it takes the state... like this the outside has the main control.

Reply:
changed according to suggestion